src/image_gen/imagegen_init.cpp

```diff
@@ -56,8 +135,82 @@ std::variant<Status, ImageGenPipelineArgs> prepareImageGenPipelineArgs(const goo
         SPDLOG_LOGGER_ERROR(modelmanager_logger, "Models path does not exist: {}", args.modelsPath);
         return StatusCode::PATH_INVALID;
     }
+    bool isNPU = false;
     if (nodeOptions.has_device()) {
-        args.device = nodeOptions.device();
+        isNPU = nodeOptions.device().find("NPU") != std::string::npos;
+        // use getListOfDevices
+        auto devicesOrStatus = getListOfDevices(nodeOptions.device());
+        if (std::holds_alternative<Status>(devicesOrStatus)) {
+            SPDLOG_LOGGER_ERROR(modelmanager_logger, "Failed to parse devices: {}", nodeOptions.device());
+            return std::get<Status>(devicesOrStatus);
+        }
+        auto devices = std::get<std::vector<std::string>>(devicesOrStatus);
+        if (devices.empty()) {
+            SPDLOG_LOGGER_ERROR(modelmanager_logger, "No valid devices found in: {}", nodeOptions.device());
+            return StatusCode::DEVICE_WRONG_FORMAT;
+        }
+
+        // allow only 1 or 3 devices
+        if (devices.size() != 1 && devices.size() != 3) {
+            SPDLOG_LOGGER_ERROR(modelmanager_logger, "Invalid number of devices specified: {}. Expected 1 or 3.", devices.size());
+            return StatusCode::DEVICE_WRONG_FORMAT;
+        }
+
+        args.device = std::move(devices);
```

Comment:
devices -> device?
Can we adjust naming?

Reply:
I see, so it's the device string. 
However you prefer, I'd just like it to match, so if we stick to device let it be 
`args.device = std::move(device);` etc.

Reply:
done